### DISCUSSION TITLE
Joyride Context and Wrapper in App.tsx

### DISCUSSION THREAD

**Discussion by shima0210** [gilbarbara/react-joyride#1068] [Q&A]
I have created this Joyride Context and Wrapper and used those in my app.tsx. I set "run: true" in StartTour function, but Joyride has not been stated. Am I missing something? I am a new user of react and joyride.

Here is my sample code

![image](https://github.com/user-attachments/assets/8b43a6e5-95a7-417b-b2fc-07803935bbf6)

![image](https://github.com/user-attachments/assets/bcbd3765-dcfb-4e6b-b0f1-8c1af6967d6a)
![image](https://github.com/user-attachments/assets/4c7df247-6890-4cde-8879-9cd389800b19)

And in App.tsx

![image](https://github.com/user-attachments/assets/06c12c2a-e758-478a-8df0-1a5ff5de1451)
![image](https://github.com/user-attachments/assets/1298a4a8-808f-4ea8-a5a0-6be7734e6b8e)

**Comment by gilbarbara** [comment]
Hey @shima0210 
Are you changing the `run` state to true?
Also, based on the code in your callback, I don't think you need to use `stepIndex`.
Setting it means you must handle the index manually, which can be tricky.

[s]

**Comment by shima0210** [comment]
Yeah, I am setting `run ` state to true in `StartTour ` function in `App.tsx`. I can remove `StepIndex ` also. But it is not working if my `App.tsx` like that. I moved all the logic to `JoyrideWrapper ` and called the wrapper in `App.tsx ` and it is working now. 

![image](https://github.com/user-attachments/assets/d848a157-a1c4-4974-b50a-c618e1686bb2)

Is it possible that `App.tsx` will have have all the conditions and call `JoyrideWrapper`

![image](https://github.com/user-attachments/assets/27698371-5fe4-447e-a9dc-e0a5e77b5563)

**Comment by gilbarbara** [comment]
Cool. You don't need a wrapper (or provider) to run Joyride.
But if your app already has too much logic, it's better to componentize.

Good luck
